src/Appwrite/Utopia/Response/Filters/V15.php

```diff
@@ -0,0 +1,556 @@
+    {
+        if (!isset($content['$permissions'])) {
+            return $content;
+        }
+
+        $read = [];
+        $write = [];
+
+        // downgrade the permissions
+        foreach ($content['$permissions'] as $permission) {
+            $permission = Permission::parse($permission);
+            $permission_value = $permission->getRole();
+            if ($permission->getIdentifier()) {
+                $permission_value .= ':' . $permission->getIdentifier();
+            }
+            if ($permission->getDimension()) {
+                $permission_value .= '/' . $permission->getDimension();
+            }
+
+            // Old type permissions meant that 'write' is equivalent to 'create', 'update' and 'delete'
+            switch ($permission->getPermission()) {
+                case Database::PERMISSION_UPDATE:
+                case Database::PERMISSION_DELETE:
+                case Database::PERMISSION_WRITE:
+                case Database::PERMISSION_CREATE:
+                    $write[$this->parseRole($permission_value)] = true;
+                    break;
+                case Database::PERMISSION_READ:
+                    $read[$this->parseRole($permission_value)] = true;
+                    break;
+            }
+        }
+
+        $content['$read'] = array_keys($read);
+        $content['$write'] = array_keys($write);
+
+        unset($content['$permissions']);
+
+        return $content;
+    }
+
+    protected function parseRole(string $role)
+    {
+        switch ($role) {
+            case Role::any()->toString():
+                return 'role:all';
+            case Role::users()->toString():
+                return 'role:member';
+            case Role::guests()->toString():
+                return 'role:guest';
+            default:
+                return $role;
+        }
+
+        return $role;
+    }
+
+    protected function parseCollection(array $content)
+    {
+        if (isset($content['documentSecurity'])) {
+            if ($content['documentSecurity']) {
+                $content['permission'] = 'document';
+            } else {
+                $content['permission'] = 'collection';
+            }
+        }
+
+        unset($content['documentSecurity']);
+        $content = $this->parsePermissions($content);
+        $content = $this->parseDatetimeAttributes($content, ['$createdAt', '$updatedAt']);
+        return $content;
+    }
+
+    protected function parsePermissionsCreatedAtUpdatedAt(array $content)
+    {
+        $content = $this->parsePermissions($content);
+        $content = $this->parseDatetimeAttributes($content, ['$createdAt', '$updatedAt']);
+        return $content;
+    }
+
+    private function parseExecution($content)
+    {
+        unset($content['stdout']);
+        $content = $this->parsePermissions($content);
+        $content = $this->parseDatetimeAttributes($content, ['$createdAt', '$updatedAt', 'startTime', 'endTime']);
+        return $content;
```

Comment:
The `time` attribute has been renamed to `duration`
https://github.com/appwrite/appwrite/pull/3801